harvester_e2e_tests/integration/test_hosts.py

```diff
@@ -67,3 +127,156 @@ def test_host_poweron_state(api_client, host_state, wait_timeout):
+        status_code, nodes_info = api_client.hosts.get()
+
+        assert status_code == 200, f"Failed to list nodes with error: {nodes_info}"
+        assert len(nodes_info['data']) > 1, "Please use multiple hosts to test vm migration"
+
+        for node_info in nodes_info.get('data', []):
+            for condition in node_info.get('status', {}).get('conditions', []):
+                if condition.get('type', "") == "Ready":
+                    assert "True" == condition.get('status', ""), (
+                        f"Node {node_info['metadata']['name']} is not ready"
+                    )
+            assert False is node_info.get('spec', {}).get('unschedulable', False), (
+                f"Node {node_info['metadata']['name']} is unschedulable"
+            )
+
+    @pytest.mark.dependency(depends=["TestHostTriggerVmMigrate::test_prerequisite"])
+    def test_maintenance_mode(self, api_client, focal_vm, wait_timeout):
+        vm_name = focal_vm['metadata']['name']
+        vm_namespace = focal_vm['metadata']['namespace']
+        code, data = api_client.vmis.get(name=vm_name, namespace=vm_namespace)
+        assert 200 == code, (
+            f"Can't get VMI {vm_namespace}/{vm_name} with error: {code}, {data}"
+        )
+        src_host = data['status']['nodeName']
+
+        code, data = api_client.hosts.maintenance_mode(src_host, enable=True)
+        assert 200 == code, (
+            f"Failed to enable maintenance mode on node {src_host} with error: {code}, {data}",
+        )
+
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, data = api_client.vmis.get(name=vm_name, namespace=vm_namespace)
+            if data.get('status', {}).get('phase', "") == "Running":
+                break
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"VM {vm_namespace}/{vm_name} can't be Running with {wait_timeout} timed out\n"
+                f"Got error: {code}, {data}"
+            )
+
+        # teardown
+        code, data = api_client.hosts.maintenance_mode(src_host, enable=False)
+        assert 200 == code, (
+            f"Failed to disable maintenance mode on node {src_host} with error: {code}, {data}",
+        )
+
+    @pytest.mark.dependency(depends=["TestHostTriggerVmMigrate::test_maintenance_mode"])
+    def test_poweroff(self, api_client, host_state, focal_vm, wait_timeout):
+        vm_name = focal_vm['metadata']['name']
+        vm_namespace = focal_vm['metadata']['namespace']
+        code, data = api_client.vmis.get(name=vm_name, namespace=vm_namespace)
+        assert 200 == code, (
+            f"Can't get VMI {vm_namespace}/{vm_name} with error: {code}, {data}"
+        )
+        src_host = data['status']['nodeName']
+
+        # update vm-force-reset-policy to 10s, so we don't need to wait 300s to force remove VM
+        code, data = api_client.settings.get("vm-force-reset-policy")
```

Comment:
This would mix the test a bit up with the setting, to make sure the setting will be tear down gracefully, we should use fixture for it.